test/OpenTelemetry.Tests.Stress/Skeleton.cs

```diff
@@ -0,0 +1,131 @@
+
+public partial class Program
+{
+    private static volatile bool bContinue = true;
+    private static volatile string output = "Test results not available yet.";
+
+    public static void Stress(int concurrency = 0)
+    {
+        if (concurrency < 0)
+        {
+            throw new ArgumentOutOfRangeException(nameof(concurrency), "concurrency level should be a non-negative number.");
+        }
+
+        if (concurrency == 0)
+        {
+            concurrency = Environment.ProcessorCount;
+        }
+
+        var statistics = new long[concurrency];
+
+        Parallel.Invoke(
+            () =>
+            {
+                Console.WriteLine($"Running (concurrency = {concurrency}), press <Esc> to stop...");
+                var bOutput = false;
+                var watch = new Stopwatch();
+                while (true)
+                {
+                    if (Console.KeyAvailable)
+                    {
+                        var key = Console.ReadKey(true).Key;
+
+                        switch (key)
+                        {
+                            case ConsoleKey.Enter:
+                                Console.WriteLine(string.Format("{0} {1}", DateTime.UtcNow.ToString("O"), output));
+                                break;
+                            case ConsoleKey.Escape:
+                                bContinue = false;
+                                return;
+                            case ConsoleKey.Spacebar:
+                                bOutput = !bOutput;
+                                break;
+                        }
+
+                        continue;
+                    }
+
+                    if (bOutput)
+                    {
+                        Console.WriteLine(string.Format("{0} {1}", DateTime.UtcNow.ToString("O"), output));
+                    }
+
+                    var cntLoopsOld = (ulong)statistics.Sum();
+                    var cntCpuCyclesOld = GetCpuCycles();
+
+                    watch.Restart();
+                    Thread.Sleep(200);
+                    watch.Stop();
+
+                    var cntLoopsNew = (ulong)statistics.Sum();
+                    var cntCpuCyclesNew = GetCpuCycles();
+
+                    var nLoops = cntLoopsNew - cntLoopsOld;
+                    var nCpuCycles = cntCpuCyclesNew - cntCpuCyclesOld;
+
+                    var nLoopsPerSecond = (double)nLoops / ((double)watch.ElapsedMilliseconds / 1000.0);
+                    var nCpuCyclesPerLoop = nLoops == 0 ? 0 : nCpuCycles / nLoops;
+
+                    output = $"Loops: {cntLoopsNew:n0}, Loops/Second: {nLoopsPerSecond:n0}, CPU Cycles/Loop: {nCpuCyclesPerLoop:n0}";
+                    Console.Title = output;
+                }
+            }, () =>
+            {
+                Parallel.For(0, concurrency, (i) =>
+                {
+                    statistics[i] = 0;
+                    while (bContinue)
+                    {
+                        statistics[i]++;
```

Comment:
Do we need change the increment and above read to atomic?

Reply:
Why?